classes/phing/types/selectors/SizeSelector.php

```diff
@@ -28,186 +28,60 @@
 class SizeSelector extends BaseExtendSelector
 {
     /**
-     * @var int $size
+     * @var float
      */
-    private $size = -1;
+    private $bytes = -1;
 
     /**
-     * @var int $multiplier
+     * @var string
      */
-    private $multiplier = 1;
+    private $value = '';
 
     /**
-     * @var int $sizelimit
+     * @var string 'less', 'equal' or 'more'.
      */
-    private $sizelimit = -1;
+    private $when = self::WHEN[0];
 
-    /**
-     * @var int $cmp
-     */
-    private $cmp = 2;
-
-    public const SIZE_KEY = "value";
-    public const UNITS_KEY = "units";
-    public const WHEN_KEY = "when";
-
-    /**
-     * @var array $sizeComparisons
-     */
-    private static $sizeComparisons = ["less", "more", "equal"];
-
-    /**
-     * @var array $byteUnits
-     */
-    private static $byteUnits = [
```

Comment:
Loosing features is not the best way. Please leave it or think about implementing this for the others as well.

Reply:
Mixing _1000_ and  _1024_  is confusing, specially because other tasks only support _1024_.

This is exactly why I created an issue first (#1480 ), I explained everything before starting coding.
A new major version will be released and this is exactly when breaking changes are done.

Reply:
Done, all these units are supported